File: Backend/UserDataService/src/main/java/com/user/dto/Goal.java
package com.user.dto;

import java.util.Date;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Goal {
	private Long goalId;
	private Double currentScore;
	private Long count;
	private boolean isAchieved;
	
	private Date creationDate;
	private String title;
	private String type;
    private String description;
    private Long targetScore;
    private Long rewardPoint; 
    private String badgeUrl;
}

File: FrontEnd/Self-Tracker/src/app/models/goal.ts
export class Goal {
    goalId: number;
    currentScore: number;
    count: number;
    isAchieved: boolean;
  
    creationDate: Date;
    title: string;
    type: string;
    description: string;
    targetScore: number;
    rewardPoint: number;
    badgeUrl: string;
  
    constructor(
      goalId: number,
      currentScore: number,
      count: number,
      isAchieved: boolean,
      creationDate: Date,
      title: string,
      type: string,
      description: string,
      targetScore: number,
      rewardPoint: number,
      badgeUrl: string
    ) {
      this.goalId = goalId;
      this.currentScore = currentScore;
      this.count = count;
      this.isAchieved = isAchieved;
      this.creationDate = creationDate;
      this.title = title;
      this.type = type;
      this.description = description;
      this.targetScore = targetScore;
      this.rewardPoint = rewardPoint;
      this.badgeUrl = badgeUrl;
    }
  }
  